### PR TITLE
Clean shuffle data

### PR DESCRIPTION
**Pull request by penghuo**
### Description
* For REPL, clean shuffle data after query result is consumed.
* For Index/MV Refresh Job, clean shuffle data after each microBatch.

### Issues Resolved
https://github.com/opensearch-project/opensearch-spark/issues/302

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spark-sql-application/src/main/scala/org/apache/spark/sql/JobOperator.scala [hunk 1]
```diff
@@ -92,6 +89,8 @@ case class JobOperator(
     try {
       // Wait for streaming job complete if no error and there is streaming job running
       if (!exceptionThrown && streaming && spark.streams.active.nonEmpty) {
+        //
```
Comment (dai-chen) [inline spark-sql-application/src/main/scala/org/apache/spark/sql/JobOperator.scala:92]: missing comment?

File: spark-sql-application/src/main/scala/org/apache/spark/sql/util/ShuffleCleaner.scala [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+/**
+ * Clean Spark shuffle data after each microBatch.
+ * https://github.com/opensearch-project/opensearch-spark/issues/302
+ */
+class ShuffleCleaner(spark: SparkSession) extends StreamingQueryListener with Logging {
+
+  override def onQueryStarted(event: StreamingQueryListener.QueryStartedEvent): Unit = {}
+
+  override def onQueryProgress(event: StreamingQueryListener.QueryProgressEvent): Unit = {
+    ShuffleCleaner.cleanUp(spark)
```
Comment (dai-chen) [inline spark-sql-application/src/main/scala/org/apache/spark/sql/util/ShuffleCleaner.scala:22]: This is triggered after each micro batch? Any impact on MV streaming job?

File: spark-sql-application/src/main/scala/org/apache/spark/sql/JobOperator.scala (hunk 1 above)
Comment (penghuo) [reply spark-sql-application/src/main/scala/org/apache/spark/sql/JobOperator.scala]: done.

**Review by dai-chen — APPROVED** [review]
Thanks for the changes!

**Pull request merged**